### PR TITLE
5_ParserUtils tests

### PR DESCRIPTION
**Pull request by pdtran3k6**
Spent some time trying to add new tests using TestNG but it seems like `./gradlew build` doesn't actually execute the tests in tony-history-server and pass everything by default, so I reverted back to JUnit.
Added some regression tests for ParserUtils.

### CODE REVIEW

**Comment by oliverhu** [comment]
why isn't junit working? if not, shall we switch all test cases to junit ? @pdtran3k6  @erwa

**Comment by pdtran3k6** [comment]
I'm cool with 2 different test frameworks. As long as the tests are actually being executed, I don't think it adds too much overhead as they both have very similar API.

**Comment by pdtran3k6** [comment]
> Gradle assumes JUnit by default. I think you need to add
> 
> ```
> test {
>   useTestNG()
> }
> ```
> for TestNG tests to run.
> 
> Ref: https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html
> 
> The root build.gradle file applies the above block to all subprojects that apply the Java plugin, but the tony-history-server subproject doesn't apply the Java plugin (only the Play plugin), so that's why I think it's not taking effect. You could update the build.gradle file to apply the above block for subprojects applying either the Java or Play plugins.

It still doesn't work for me. We should sync up tmr.

**Comment by erwa** [comment]
Let's sync up about the JUnit/TestNG offline.

File: tony-history-server/test/utils/TestParserUtils.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  @Test
+  public void testIsValidHistFileName_false() {
+    String fileName1 = "application123-1-1-user1-FAILED.jhist";
+    String fileName2 = "job123-1-1-USER-SUCCEEDED.jhist";
+    String jobRegex = "job\\d+";
+
+    assertFalse(ParserUtils.isValidHistFileName(fileName1, jobRegex));
+    assertFalse(ParserUtils.isValidHistFileName(fileName2, jobRegex));
```
[the remark this answers is not in the document]
Comment (pdtran3k6) [reply tony-history-server/test/utils/TestParserUtils.java]: 👍

**Pull request merged**